Sources/StreamChat/Database/DataStore.swift

```diff
@@ -4,6 +4,89 @@
 
 import Foundation
 
+protocol EntityCache {
+    func get(userID: String) -> UserDTO?
+    func set(user: UserDTO)
+
+    func get(channelCID: String) -> ChannelDTO?
+    func set(channel: ChannelDTO)
+
+    func get(memberID: String) -> MemberDTO?
+    func set(member: MemberDTO, channelId: ChannelId)
+
+    func get(messageID: String) -> MessageDTO?
+    func set(message: MessageDTO)
+
+    func flush()
+}
+
+class InMemCache: EntityCache {
```

Comment:
The idea looks promising. We have to be careful about the API, doesn't seem scalable atm

Reply:
cache lives for 1 db context write, we dont grow past that